micronota/db/uniref.py

```diff
@@ -0,0 +1,210 @@
+UniRef90 and UniRef50 clusters) of sequences from the UniProtKB
+and selected UniParc records, in order to obtain complete coverage of sequence
+space at several resolutions (100%, >90% and >50%) while hiding redundant
+sequences (but not their descriptions) from view.
+
+Release
+-------
+:version: 2016_01
+:date:    01/20/2016
+:link:    ftp://ftp.uniprot.org/pub/databases/uniprot/relnotes.txt
+
+Files
+-----
+* UniRef files
+
+  UniRef fasta files clustered at the similarity levels of 50, 90 or 100.
+  The UniRef100 identifier is generated by placing "UniRef100_" prefix
+  before the UniProtKB accession number or UniParc identifier of the
+  representative UniProtKB or UniParc entry. It is similarly done for
+  UniRef50 or UniRef90.
+
+  * uniref50.fasta.gz [#]_
+
+  * uniref90.fasta.gz [#]_
+
+  * uniref100.fasta.gz [#]_
+
+* UniProtKB files
+
+  The .dat.gz files contain the UniProtKB records in variant format of EMBL.
+  The records are divided into taxa groups [#]_.
+
+  * uniprot_sprot.dat.gz
+
+  * uniprot_trembl.dat.gz
+
+Reference
+---------
+.. [#] http://www.uniprot.org/help/uniparc
+.. [#] http://www.uniprot.org/help/uniprotkb
+.. [#] http://www.ncbi.nlm.nih.gov/pubmed/17379688
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref50/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref90/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/uniref/uniref100/README
+.. [#] ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/taxonomic_divisions/README
+'''
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from os.path import join, expanduser
+from sqlite3 import connect
+import shutil
+
+from skbio import read, Sequence
+
+from ..util import _overwrite
+
+
+def prepare_db(out_d, prefix='uniref100', force=False,
+               server='ftp.uniprot.org',
+               path='pub/databases/uniprot/uniref/uniref100',
+               fasta='uniref100.fasta.gz'):
+    '''Prepare reference database for UniRef.
+
+    Notes
+    -----
+    This function creates the following files:
+
+    * ``uniref100_sprot_archaea.dmnd``
+    * ``uniref100_sprot_bacteria.dmnd``
+    * ``uniref100_sprot_viruses.dmnd``
+    * ``uniref100_sprot_other.dmnd``
+    * ``uniref100_trembl_archaea.dmnd``
+    * ``uniref100_trembl_bacteria.dmnd``
+    * ``uniref100_trembl_viruses.dmnd``
+    * ``uniref100_trembl_other.dmnd``
+    * ``uniref100_other.dmnd``
+
+    * ``uniprotkb.db``
+    * ``uniref_100_90_50.db``: the map between UniRef100, UniRef90, and UniRef50.
+    '''
+
+    prepare_metadata(join(expanduser('~'), 'uniref', 'uniprot_sprot.dat.gz'),
```

Comment:
Really not a fan of hard coding these values in.  Can we pass these in as parameters?